samples/sqlservices/src/Providers/connectionProvider.ts

```diff
@@ -0,0 +1,87 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+
+export const ProviderId: string = 'TESTPROVIDER';
+
+/**
+ * This class implements the ConnectionProvider interface that allows users to connect to the data services, this will be used by various features in ADS, e.g. connection dialog and query editor.
+ * Use this approach if the data service has native JavaScript SDK available, otherwise refer to mssql extension on how to leverage the dataprotocol client which has JSON/RPC support.
+ */
+export class ConnectionProvider implements azdata.ConnectionProvider {
+	private onConnectionCompleteEmitter: vscode.EventEmitter<azdata.ConnectionInfoSummary> = new vscode.EventEmitter();
+	onConnectionComplete: vscode.Event<azdata.ConnectionInfoSummary> = this.onConnectionCompleteEmitter.event;
+
+	private onIntelliSenseCacheCompleteEmitter: vscode.EventEmitter<string> = new vscode.EventEmitter();
+	onIntelliSenseCacheComplete: vscode.Event<string> = this.onIntelliSenseCacheCompleteEmitter.event;
+
+	private onConnectionChangedEmitter: vscode.EventEmitter<azdata.ChangedConnectionInfo> = new vscode.EventEmitter();
+	onConnectionChanged: vscode.Event<azdata.ChangedConnectionInfo> = this.onConnectionChangedEmitter.event;
+
+	connect(connectionUri: string, connectionInfo: azdata.ConnectionInfo): Promise<boolean> {
```

Comment:
I feel like if we're making samples we should strive to have comments explaining all the important bits. Like for example here it would probably be good to say "This is where your code to actually make a connection should go"

Reply:
I am actively working with a partner right now, I will be making changes and add comments.